public/app/features/explore/Logs.tsx

```diff
@@ -72,13 +74,31 @@ interface State {
   showLabels: boolean;
   showTime: boolean;
   wrapLogMessage: boolean;
+  logsSortOrder: LogsSortOrder | null;
+  disabledSortButton: boolean;
 }
 
 export class Logs extends PureComponent<Props, State> {
   state = {
     showLabels: store.getBool(SETTINGS_KEYS.showLabels, false),
     showTime: store.getBool(SETTINGS_KEYS.showTime, true),
     wrapLogMessage: store.getBool(SETTINGS_KEYS.wrapLogMessage, true),
+    logsSortOrder: null,
+    disabledSortButton: false,
+  };
+
+  onChangeLogsSortOrder = () => {
+    this.setState({ disabledSortButton: true });
+    // we are using setTimeout here to make sure that disabled button is rendered before the rendering of reordered logs
+    setTimeout(() => {
```

Comment:
To be really clean, you'd track both timeouts in component variables and clear them before component unmount.